packages/adapters/watcher/src/types.ts

```diff
@@ -41,8 +53,51 @@ export abstract class Verifier {
     }
     return connext;
   }
+
+  public getRootManagerDeployment(chainId: number): { address: string; abi: any } {
+    const rootManager = getDeployedRootManagerContract(chainId, this.context.isStaging ? "Staging" : "");
+    if (!rootManager) {
+      // TODO: Custom errors for package
+      throw new Error(`RootManager deployment not found for chain ${chainId}!`);
+    }
+    return rootManager;
+  }
+
+  public getMerkleTreeManagerOfRootManagerDeployment(chainId: number): { address: string; abi: any } {
+    const merkleTreeManager = getDeployedMerkleTreeManagerOfRootManagerContract(
+      chainId,
+      this.context.isStaging ? "Staging" : "",
+    );
+    if (!merkleTreeManager) {
+      // TODO: Custom errors for package
+      throw new Error(`MerkleManager deployment not found for chain ${chainId}!`);
+    }
+    return merkleTreeManager;
+  }
+
+  public getSpokeConnectorDeployment(chainId: number): { address: string; abi: any } {
+    const protocol = this.context.isStaging ? MESSAGING_PROTOCOL_CONFIGS.testnet : MESSAGING_PROTOCOL_CONFIGS.mainnet;
+    const connectorName = getConnectorName(protocol, chainId);
+    const spokeConnector = getDeployedSpokeConnecterContractByName(
+      chainId,
+      connectorName,
+      this.context.isStaging ? "Staging" : "",
+    );
+    if (!spokeConnector) {
+      // TODO: Custom errors for package
+      throw new Error(`SpokeConnector deployment not found for chain ${chainId}!`);
+    }
+    return spokeConnector;
+  }
 }
 
+export type SwitchResponse = {
+  domain: number;
+  switched: boolean;
+  error: any;
+  relevantTransaction: ethers.providers.TransactionResponse | string;
```

Comment:
should be a transaction receipt right? if sent using the `chainservice`, it would be a receipt as the bumping / monitoring is handled internally to that class